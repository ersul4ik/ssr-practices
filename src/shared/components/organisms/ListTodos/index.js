import React from 'react'
import { connect } from 'react-redux'

import { deleteTodo, fetchTodos, tasksFetchDataSuccess} from '../../../actions'

export const mapDispatchToProps = (dispatch) => {
  return {
    deleteTodo: id => dispatch(deleteTodo(id)),
    fetchTodos: () => dispatch(fetchTodos()),
  }
}

class ListTodos extends React.Component {
  componentWillMount() {
    this.props.fetchTodos()
  }

  render() {
    const todos = this.props.todos
    return (
      <div>
        <ul className="list-group">
          {todos.map(todo =>
            (<li key={todo.task_id} name="nameTask" className="list-group-item">
              {todo.description}
              <button
                type="button"
                className="close right"
                aria-label="Close"
                onClick={() => this.props.deleteTodo(todo.task_id)}
                style={{ marginLeft: '5px' }}
              >
                <span aria-hidden="true">&times;</span>
              </button>
            </li>))}
        </ul>
      </div>

    )
  }
}

export default connect(null, mapDispatchToProps)(ListTodos)
