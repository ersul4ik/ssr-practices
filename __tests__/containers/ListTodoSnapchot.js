import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import ListTodos from '../../src/shared/containers/Todos';

const mockStore = configureMockStore();

describe('Link', () => {
    let wrapper, store;

    beforeEach(() => {
        const initialState = {
            todos: { id: 1, description: 'first' }
        };
        store = mockStore(initialState);
        wrapper = shallow(
            <Provider store={store}>
                <ListTodos todos={store} />
            </Provider>
        );
    });
    it('should render correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});