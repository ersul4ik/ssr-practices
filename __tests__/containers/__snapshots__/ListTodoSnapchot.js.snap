// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
<Connect(Connect(ListTodos))
  todos={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
